lib/Doctrine/DBAL/Driver/AbstractFbIbDriver.php

```diff
@@ -0,0 +1,154 @@
+<?php
+
+/*
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * This software consists of voluntary contributions made by many individuals
+ * and is licensed under the MIT license. For more information, see
+ * <http://www.doctrine-project.org>.
+ */
+
+namespace Doctrine\DBAL\Driver;
+
+/**
+ * Abstract Firebird/Interbase driver.
+ *
+ * <b>This Driver/Platform is in Beta state</b>
+ * 
+ * @author Andreas Prucha, Helicon Software Development <prucha@helicon.co.at>
+ */
+abstract class AbstractFbIbDriver implements \Doctrine\DBAL\Driver, ExceptionConverterDriver
+{
+
+    /**
+     * Driver options
+     * 
+     * @var type 
+     */
+    protected $driverOptions = array();
+
+    /**
+     * Driver option for additional platform configuration
+     * @see Abstract
+     */
+    const DRIVER_OPTION_PLATFORM_OPTIOS = 'FbIbPlatformOptions';
+    
+    /**
+     * {@inheritDoc}
+     * 
+     * This function does not actually 
+     * 
+     * @param array $params
+     * @param type $username
+     * @param type $password
+     * @param array $driverOptions
+     */
+    public function connect(array $params, $username = null, $password = null, array $driverOptions = array())
```

Comment:
It seems very weird to me that it does not return a connection as defined by the interface